src/libsync/discovery.cpp

```diff
@@ -373,6 +373,20 @@ void ProcessDirectoryJob::processFile(PathTuple path,
                               << " | e2eeMangledName: " << dbEntry.e2eMangledName() << "/" << serverEntry.e2eMangledName
                               << " | file lock: " << localFileIsLocked << "//" << serverFileIsLocked;
 
+    if(serverEntry.locked == SyncFileItem::LockStatus::LockedItem) {
+        const auto lockExpirationTime = serverEntry.lockTime + serverEntry.lockTimeout;
+        const auto timeRemaining = QDateTime::currentDateTime().secsTo(QDateTime::fromSecsSinceEpoch(lockExpirationTime));
+        const auto timerInterval = qMax(5ll, timeRemaining);
+
+        qCInfo(lcDisco) << "Will re-check lock status for:" << path._original << "in:" << timerInterval << "seconds.";
+
+        // lockTimeout is in secs, QTimer takes msecs
+        QTimer::singleShot(timerInterval * 1000, this, [this, path]{
+            qCInfo(lcDisco) << "Lock for file:" << path._original << "has expired. Rescanning.";
+            startAsyncServerQuery();
```

Comment:
what do you want to achieve here ?
that line of code will trigger a server query for the state of files at a point were the code is supposed to already have all information about the state of the file
don't you want to trigger a new sync run to apply any eventual changes from server ?

Reply:
Apologies if I misunderstood the function of this method -- what I want to do is get the lock state from the server instead of applying changes to the lock state of the file locally without checking for changes from the server first. So this timer is intended to schedule a sync run after the lock timer runs out

Reply:
Okay, I have changed it to schedule a sync run